drivers/flash/soc_flash_stm32f4.c

```diff
@@ -26,20 +25,19 @@
 
 #include <flash_registers.h>
 
-static const struct flash_map {
-      uint32_t start;
-      uint32_t len;
-      uint16_t id;
+static struct flash_map {
+	uint32_t start;
+	const uint32_t len;
+	const uint16_t id;
 } sector [] = {
-	/* TODO: investigate why sector 0 is not accessible */
-	{ .start = 0x08000000 , .len = KB(16),   .id = 0x0000 },
-	{ .start = 0x08004000 , .len = KB(16),   .id = 0x0008 },
-	{ .start = 0x08008000 , .len = KB(16),   .id = 0x0010 },
-	{ .start = 0x0800c000 , .len = KB(16),   .id = 0x0018 },
-	{ .start = 0x08010000 , .len = KB(64),   .id = 0x0020 },
-	{ .start = 0x08020000 , .len = KB(128),  .id = 0x0028 },
-	{ .start = 0x08040000 , .len = KB(128),  .id = 0x0030 },
-	{ .start = 0x08060000 , .len = KB(128),  .id = 0x0038 },
+	{ .len = KB(16),   .id = 0x0000 },
+	{ .len = KB(16),   .id = 0x0008 },
+	{ .len = KB(16),   .id = 0x0010 },
+	{ .len = KB(16),   .id = 0x0018 },
+	{ .len = KB(64),   .id = 0x0020 },
+	{ .len = KB(128),  .id = 0x0028 },
+	{ .len = KB(128),  .id = 0x0030 },
+	{ .len = KB(128),  .id = 0x0038 },
```

Comment:
Would you mind protecting this with CONFIG_SOC_STM32F401RE ? Depending on the MCU rev, the size of the flash will differ.

This driver can probably also be generic, like soc_flash_stm32.c, as done with the clock/gpio/etc drivers.